proposals/0000-config-finalization.md

```diff
@@ -0,0 +1,160 @@
+- Start Date: 2022-03-22
+- Reference Issues: [[Roadmap] Astro v1.0](https://github.com/withastro/rfcs/discussions/1)
+- Implementation PR: <!-- leave empty -->
+
+# Summary
+
+Astro's configuration file is something we'd like to clean up prior to a v1.0 Beta release. Currently, it is not evident which specific options should go in which area, leading to user confusion and ocassional duplication.
+
+This RFC proposes a clean-up of the existing Astro config file format for consistency and clarity.
+
+# Example
+
+When this RFC is accepted, a config file for Astro might look something like this:
+
+```js
+import { defineConfig } from 'astro';
+import preact from '@astrojs/preact';
+
+export default defineConfig({
+    log: 'info',
+
+    rootDir: '.',
```

Comment:
nit: I'm a big fan of keeping abbreviations to a minimum, so I'd prefer `rootDirectory` over `rootDir`. I expressed this earlier and think I'm outvoted here, and this is far from a hill I'll die on, but consider this me registering my opinion :)

Also, given my other comments here I do feel a responsibility to call out that Vite uses `Dir` for `publicDir`, so I do see the value in matching there.

Reply:
I don't disagree but looking at lots of other projects, I concluded that `Dir` was the unspoken standard here.

Reply:
It is worth calling out that Vite has `root` and then everything else (`publicDir`) gets the `Dir` suffix. Seemed confusing to have two different ways to name a directory option.

Reply:
Yea, If we feel comfortable not matching Vite in some cases, I think that means we probably don't feel that strongly about matching Vite in other cases either. IDK, not a hill I'll die on but this does strike me as a bit odd as I try to weigh these objectively.